statwrap/ux.py

```diff
@@ -63,22 +61,20 @@ def createDF(self):
         if file_extension == 'csv':
             df = pd.read_csv(content_stream)
         elif file_extension in {'xls', 'xlsx', 'xlsm', 'xlsb'}:
-            df = pd.read_excel(content_stream)
+            df = pd.read_excel(content_stream, engine='openpyxl' if file_extension == 'xlsx' else None)
         elif file_extension in {'odf', 'ods', 'odt'}:
             df = pd.read_excel(content_stream, engine='odf')
         else:
-            raise ValueError("Unsupported file extension")
+            raise ValueError("Unsupported file extension")  # unreachable
 
         return df
 
-    def uploadData(self):
-        def on_submit(button):
-            df = self.createDF()
-            ipython = get_ipython()
-            ipython.user_global_ns[self.variable_name] = df
-            display(df.head())
-        
-        self.submit_button.on_click(on_submit)
-        
-        if self.auto_display:
-            display(self.uploader, self.submit_button)
\ No newline at end of file
+    def _on_submit(self, button):
+        df = self._create_df()
+        ipython = get_ipython()
+        ipython.user_global_ns[self.variable_name] = df
+        display(df.head())
+
+def convenience_display(widget):
```

Comment:
we can hold off on this. I would cut this function and later we can create a function like `data_upload` that creates and displays a DataFileUploader for a basic user.